sdk/core/azure-core/inc/azure/core/azure_assert.hpp

```diff
@@ -44,13 +44,10 @@
 
 #include <cassert>
 
-#define AZURE_ASSERT(exp) assert((exp))
 #define AZURE_ASSERT_MSG(exp, msg) assert(((void)msg, (exp)))
 
 #endif
 
 [[noreturn]] void AzureNoReturnPath(std::string const& msg);
 
-#define AZURE_ASSERT_FALSE(exp) AZURE_ASSERT(!(exp))
 #define AZURE_UNREACHABLE_CODE AzureNoReturnPath("unreachable code!")
-#define AZURE_NOT_IMPLEMENTED AzureNoReturnPath("not implemented code!")
```

Comment:
These ASSERTS are expected to be used by Storage.  Wait until @Jinming-Hu replace std::about to use AZURE_ASSERT.

Storage code will benefit of AZURE_ASSERT_FALSE because they check false conditions.

Reply:
Sounds good, will wait until after storage has used the asserts.

Reply:
> These ASSERTS are expected to be used by Storage. Wait until @Jinming-Hu replace std::about to use AZURE_ASSERT.

I'm going to replace `std::abort` with `AZURE_ASSERT`.

> Storage code will benefit of AZURE_ASSERT_FALSE because they check false conditions.

I'm not a big fan of this. I'd prefer `AZURE_ASSERT(!condition);`